workers/pipeline/pipeline.go

```diff
@@ -120,19 +120,28 @@ func (ap *ActivePipelines) Append(p gaia.Pipeline) {
 }
 
 // Update updates a pipeline at the given index with the given pipeline.
-func (ap *ActivePipelines) Update(index int, p gaia.Pipeline) {
+func (ap *ActivePipelines) Update(index int, p gaia.Pipeline) error {
 	ap.Lock()
 	defer ap.Unlock()
 
+	if index >= len(ap.Pipelines) || index < 0 {
+		return fmt.Errorf("invalid index for len %d. index was: %d", len(ap.Pipelines), index)
+	}
 	ap.Pipelines[index] = p
+	return nil
 }
 
 // Remove removes a pipeline at the given index from ActivePipelines.
-func (ap *ActivePipelines) Remove(index int) {
+func (ap *ActivePipelines) Remove(index int) error {
 	ap.Lock()
 	defer ap.Unlock()
 
+	l := len(ap.Pipelines)
+	if index >= l || index+1 > l {
```

Comment:
```suggestion
	if index >= l || index < 0 {
```

Reply:
Hey. :) 

So, yeah, I left out 0 that's a good call. However, I'll tell you why `index+1 > l` needs to be there. :D It's a mind bender so... 

This is the line we are protecting:
`ap.Pipelines = append(ap.Pipelines[:index], ap.Pipelines[index+1:]...)`
if l == 0:
The first part of append will fail.
if l == 1 and index = 1:
The first part will work because slice postfix is not inclusive.
But the second part, the `index+1` will fail because that one is inclusive. And 2 is not a valid index to look from.